### PR TITLE
Fix bugs in KML parsing with the new file

### PR DESCRIPTION
**Pull request by bbesmanoff**
Does several things, including not crashing and also displaying the
tour route on the map

### CODE REVIEW

**Comment by thenaterhood** [comment]
I'm not sure I understand the logic in this, but I also know it's basically the logic that was in use before that worked. Looks fine.

**Comment by bbesmanoff** [comment]
I can probably clarify the comment.  Basically, check if there is extended data.  If there is, use that.  Otherwise, run the old logic.

Ramble time: the old logic had the description stuff, which works for kml files that have `[AREA]`, but this only does what is necessary to put a basic element on the map.

File: third_party_libraries/kml.js [hunk 1]
```diff
@@ -440,7 +451,7 @@ L.Util.extend(L.KML, {
 			for (i = 0; i < el.length; i++) {
 				switch(tag) {
 					case 'LineString':
-  						var layer = this.parseLineString(el[i], xml, options);
+						var l = this.parseLineString(el[i], xml, options);
```
Comment (cman131) [inline third_party_libraries/kml.js]: If possible I'd prefer this to say "layer" as L is kind of a crappy variable name

Comment (bbesmanoff) [inline third_party_libraries/kml.js]: I definitely agree with that.  I was going with the style match for the rest of the function.

Comment (cman131) [inline third_party_libraries/kml.js]: meh alright, I guess it makes sense to keep from confusing with the later "layer" variable

**Comment by cman131** [comment]
Approved :european_post_office: :panda_face: :+1:

**Comment by thenaterhood** [comment]
:european_post_office:

**Pull request merged**